packages/react/src/analytics/integrations/GA4/GA4.ts

```diff
@@ -148,9 +150,36 @@ class GA4 extends integrations.Integration<GA4IntegrationOptions> {
       true,
     );
 
+    const dataLayer = get(
+      this.options,
+      OPTION_DATA_LAYER_NAME,
+      DEFAULT_DATA_LAYER_NAME,
+    );
+
+    googleConsentMode.addIntegration(this.loadData.consent, dataLayer);
     this.loadGtagScript(options);
   }
 
+  /**
+   * Sets the consent object. This method is called by analytics whenever the consent
+   * changes, so there's no need to validate if it has changed or not.
+   *
+   * @param consent - Object to be written on the dataLayer.
+   *
+   * @returns This allows chaining of class methods.
+   */
+  override setConsent(this: this, consent: ConsentData): this {
+    const dataLayer = get(
+      this.options,
+      OPTION_DATA_LAYER_NAME,
+      DEFAULT_DATA_LAYER_NAME,
+    );
+
+    googleConsentMode.setConsents(consent, dataLayer);
```

Comment:
Why no call `setConsentForDataLayer`? Also, the arguments should be reversed, i.e., dataLayer should be first.

Reply:
SetConsent method, work in two ways: with or without datalayer property.

If datalayer property set, then googleConsentMode should sent only to this datalayer. If property is empty, the method dispatch for each datalayer added do google Consent Mode.


For example, If I want to send manually consents to gtm and google analytics, I could by send this parameter undefined.